### PR TITLE
Contributor patch

### PR DESCRIPTION
**Pull request by sgoggins**
This vastly improves the performance and reliability of the contributor worker.

### CODE REVIEW

**Review by gabe-heim — APPROVED** [review]
LGTM, after @ccarterlandis reviews too and double checks everything I think we are good to merge.

**Review by ccarterlandis — CHANGES_REQUESTED** [review]
Please remove all the GSoC related documentation and code.

**Comment by sgoggins** [comment]
> Please remove all the GSoC related documentation and code.

Done

File: .gitignore [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
 # Config
 augur.config.json
-!default.config.json
 !frontend.config.json
 env.txt
 docker_env.txt
 pyenv.txt
 augur_export_env.sh
 .DS_Store
+*.config.json
```
Comment (sgoggins) [inline .gitignore:10]: Should these lines go?

File: augur/config.py [hunk 2]
```diff
@@ -112,6 +112,40 @@
                     ],
                     "model": "releases",
                     "repo_group_id": 0
+                },
```
Comment (sgoggins) [inline augur/config.py:115]: These are going to be needed when we deploy the ML workers, and I would rather have them included in new installations so people don't need to update their configs when we release these workers.

File: augur/config.py [hunk 3]
```diff
@@ -183,7 +217,36 @@
                 "port": 51200,
                 "switch": 1,
                 "workers": 1
-            }
+            },
```
Comment (sgoggins) [inline augur/config.py:220]: These are needed when we relesae these workers, and updating configs once installed is another step. For now they are harmless.

File: tests/test_workers/test_gitlab_worker.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import pytest
```
Comment (sgoggins) [inline tests/test_workers/test_gitlab_worker.py:1]: the gitlab worker is released already and works.  Leaving this test.

File: workers/github_worker/github_worker.py [hunk 5]
```diff
@@ -43,7 +43,7 @@ def issues_model(self, entry_info, repo_id):
         """ Data collection function
         Query the GitHub API for issues
         """
-        
+        return self.config
```
Comment (ccarterlandis) [inline workers/github_worker/github_worker.py:46]: This is going to break the GitHub worker (again). Please delete this line.

File: workers/gitlab_issues_worker/gitlab_issues_worker.py [hunk 6]
```diff
@@ -49,28 +30,10 @@ def __init__(self, config={}):
         self.tool_source = 'Gitlab API Worker'
         self.tool_version = '0.0.0'
         self.data_source = 'GitLab API'
+        self.platform_id = 25150
```
Comment (ccarterlandis) [inline workers/gitlab_issues_worker/gitlab_issues_worker.py:33]: Why is this hardcoded?

**Review by ccarterlandis — CHANGES_REQUESTED** [review]
There are still GSoC changes that need to be undone.

File: augur/config.py (hunk 2 above)
Comment (sgoggins) [reply augur/config.py:115]: It does if you are the person maintaining 50 instances of Augur and don't want to have to manually edit the config when those are released 50 times. :)

File: augur/repos.csv [hunk 7]
```diff
@@ -0,0 +1 @@
+ 1,https://gitlab.com/cerfacs/batman.git
```
[the remark this answers is not in the document]
Comment (sgoggins) [reply augur/repos.csv:1]: Its a rich repository that is useful for demoing the ML workers. I would like to keep it.

File: repos.csv [hunk 8]
```diff
@@ -0,0 +1 @@
+ 1, https://gitlab.com/cerfacs/batman.git
```
[the remark this answers is not in the document]
Comment (sgoggins) [reply repos.csv:1]: Its a rich repository that is useful for demoing the ML workers. I would like to keep it.

File: workers/gitlab_issues_worker/gitlab_issues_worker.py (hunk 6 above)
Comment (sgoggins) [reply workers/gitlab_issues_worker/gitlab_issues_worker.py:33]: There is not currently a standard way of specifying that variable, and the gitlab_issue worker needs it. Can @ccarterlandis or @gabe-heim  fix this?

File: augur/config.py (hunk 3 above)
Comment (sgoggins) [reply augur/config.py:220]: Just leave them please.

File: repos.csv (hunk 8 above)
[the remark this answers is not in the document]
Comment (sgoggins) [reply repos.csv:1]: leave it.

**Review by sgoggins — COMMENTED** [review]
I've done everything asked, except where it would create a *ton* of additional work for me. 

If there's any breaking change you identify, please fix it and merge this PR.

**Review by ccarterlandis — APPROVED** [review]

**Pull request merged**